src/hotspot/share/opto/memnode.cpp

```diff
@@ -485,23 +494,30 @@ bool MemNode::all_controls_dominate(Node* dom, Node* sub) {
       if (!n->is_CFG() && n->pinned()) {
         // Check only own control edge for pinned non-control nodes.
         n = n->find_exact_control(n->in(0));
-        if (n == nullptr || n->is_top())
+        if (n == nullptr || n->is_top()) {
+          dead_code_flag = true;
           return false; // Conservative answer for dead code
+        }
         assert(n->is_CFG(), "expecting control");
         dom_list.push(n);
       } else if (n->is_Con() || n->is_Start() || n->is_Root()) {
         only_dominating_controls = true;
       } else if (n->is_CFG()) {
-        if (n->dominates(sub, nlist))
+        bool dead_code;
+        if (n->dominates(sub, nlist, dead_code)) {
           only_dominating_controls = true;
-        else
+        } else {
+          if (dead_code) dead_code_flag = true;
```

Comment:
A few comments about this line:
- You need curly braces for the body
- We do not do implicit null-checks: you should always do `dead_code != null`. But why are you even checking this here? In all other cases you would just set the `dead_code_flag`, and if `dead_code` then this would go to the `dummy_flag`, right?

Reply:
This line is even sillier than I thought once I understood your comment, please allow me to explain:

In this scope, `dead_code` is not the pointer argument of the method, it's just a local variable with the same name. I declared it here to accept the return value of `Node::dominates`, and if it's true, we should set the pointer argument to true to indicate that we have encountered a dead control flow.

This is a bad practice as the name of the variable can cause confusion. Sorry for writing this🤦, I've removed the relevant code in the latest commit.